bluefog/common/tensor_queue.h

```diff
@@ -82,13 +82,36 @@ class FusionBufferManager {
                           std::function<void()> on_start_init,
                           std::function<void()> on_end_init);
 
+  // Initializes a buffer of the given threshold size times MPI size if not already cached.
```

Comment:
Note: There is one constraint here actually. We need WeightBuffer is always larger than (size-1)*fusion Buffer since we don't want to tensor being able to put into the fusion buffer but not able to put into weightbuffer